leader/leader.go

```diff
@@ -174,6 +174,7 @@ func Become(ctx context.Context, lockName string, opts ...Option) error {
 				leaderPod := &corev1.Pod{}
 				key = crclient.ObjectKey{Namespace: ns, Name: existingOwners[0].Name}
 				err = config.Client.Get(ctx, key, leaderPod)
+				leaderPod.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Pod"))
```

Comment:
Ditto to setting GVK - what are we doing that requires it? Can we use a Scheme with the kinds registered?

Reply:
I made this change to satisfy the failing test that was expecting this info to be set on the object after the `config.Client.Get()` call but it wasn't. Didn't think to look into setting the Scheme - will give that a shot. 

I was trying to go for a minimal change to existing behavior and follow precedence of manually setting stuff after a Get call since I _believe_ this package has been for a while intended to be deprecated (see #117 - I could be misinterpreting this though) and the intent was to get the kubernetes packages bumped.

Reply:
No scheme changes were necessary - I totally missed it, but looked like the tests were using a fake client. Setting the appropriate fields on the objects in the fake client resolved the issue.